extensions/resteasy-classic/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/NoMessageBodyWriterFoundFailureExceptionMapper.java

```diff
@@ -0,0 +1,27 @@
+package io.quarkus.resteasy.runtime;
+
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+import org.jboss.resteasy.core.NoMessageBodyWriterFoundFailure;
+
+@Provider
+public class NoMessageBodyWriterFoundFailureExceptionMapper implements ExceptionMapper<NoMessageBodyWriterFoundFailure> {
+
+    @Override
+    public Response toResponse(NoMessageBodyWriterFoundFailure exception) {
+        return Response.status(exception.getErrorCode()).header(HttpHeaders.CONTENT_TYPE, MediaType.TEXT_PLAIN)
+                .entity(determineMessage(exception)).build();
+    }
+
+    private String determineMessage(NoMessageBodyWriterFoundFailure exception) {
+        if ((exception.getMessage() != null) && exception.getMessage().contains(MediaType.APPLICATION_JSON)) {
+            return exception.getMessage()
+                    + ". Consider adding one of the following extensions: 'quarkus-resteasy-jackson', 'quarkus-resteasy-jsonb'";
```

Comment:
We already do something about it here: https://github.com/quarkusio/quarkus/blob/main/extensions/resteasy-classic/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyCommonProcessor.java#L220 . And also for Mutiny types.

Not sure if it's worth adding more?

Reply:
Hm... I wasn't aware of that. Yeah, it's probably a better approach. Let me check

Reply:
I actually do see `2021-08-30 12:55:06,338 WARN  [io.qua.res.com.dep.ResteasyCommonProcessor] (build-4) Quarkus detected the need of REST JSON support but you have not provided the necessary JSON extension for this. You can visit https://quarkus.io/guides/rest-json for more information on how to set one.
` in the logs now that you mention it.

I'll leave it up to you and @maxandersen to decide whether this provides additional benefit or not.

Reply:
The issue is that the error you get when using curl or browser to access the resource does not tell you what is the cause of error.

At least in devmode we should be able to show more informative error page.

Reply:
What I did is indeed dev-mode only